### PR TITLE
profile page && styled-components grid

### LINKED ISSUE #988: [Affilite] Implement page design
# Summary

Implement the design that's the outcome of the task #958

# Requirements

- Design implemented on `Profile` page

### PR DESCRIPTION
**Pull request by maria-vslvn**
# Summary
Fixes #988 

*High-level description of what your changes are accomplishing*
Profile view implemented
Custom common components created in order to have less small components and avoid duplication of code. Check it please, maybe we could use it somehow or find a better decision?

  # To Test

1. Visit profile page, please notice: all data are mocked
2. as new global styles were applied, it need testing to

### CODE REVIEW

**Comment by elena-zh** [comment]
Hey @maria-vslvn , great job. 

Some notes: 
1. bottom elements are missing on newly designed pages. It would be nice to add them 
image

2. It seems to me, that background is blurred more than on implemented pages: 
image

3. Also, I have a strange impression when I switch between Swap and profile pages, and see a clear background picture and a blurred one. May be we could leave the same background for all pages.. WDYT, @alfetopito ?
https://drive.google.com/file/d/1zey7wouL_qzBK1wHwinJQlCRUaFj4X6I/view

4. I see a lot of 'style' errors in the console
image

5. I think, we should keep 'Copy' icon right after a referral link and not to move it to the next row
image

6. Copy icon looks different as on other places in the app. Its behavior is also different:
image
image
image

**Comment by elena-zh** [comment]
Mobile: 
1. There is no an announcement at the top of the profile page: area seems to be empty in a mobile device
image 
image

2. On the mock-up all profile page elements are visible without scrolling (case about making elements smaller)
image 
image

**Comment by elena-zh** [comment]
@alfetopito , @alongoni , 
Global question: 
we have different bg color on existing content section and on the mock up. Moreover, the section looks transparent a bit on the mock-up. Do we need to apply all these changes to existing pages?
image

File: src/assets/styles/styled.ts [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+`
+
+export const Column = styled.div<Partial<CSS.Properties & ColumnProps>>`
+  padding: ${(props) => props.padding || '0 0.625rem'};
+  max-width: ${(props) => (props.xs !== 'hidden' && props.xs ? props.xs + '%' : '100%')};
+  flex: 0 0 ${(props) => (props.xs !== 'hidden' && props.xs ? props.xs + '%' : '100%')};
+  ${(props) => props.flexCol && 'display:flex;flex-direction:column;'}
+  ${(props) => props.flexRow && 'display:flex;flex-wrap:wrap; align-items:center'}
+  ${(props) => props.stretch && 'height:100%;'}
+  @media (max-width:767px) {
```
Comment (alfetopito) [inline src/assets/styles/styled.ts:205]: For this we usually have these media queries https://github.com/gnosis/cowswap/blob/b366fa833887f6a86dacf93f425e2dc6f250c811/src/theme/styled.ts#L70

File: src/components/icons/index.tsx [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    </SvgWrap>
+  )
+}
+
+QuestionIcon.propTypes = {
+  color: PropTypes.string,
+  width: PropTypes.string,
+}
+
+export const CopyIcon: FunctionComponent<IconProps> = ({ color, width = '16' }): React.ReactElement => {
```
Comment (alfetopito) [inline src/components/icons/index.tsx:57]: As Elena mentioned, there's already another copy icon already used in the app, I would prefer to re-use that to keep the style consistent.

File: src/components/icons/index.tsx [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    </SvgWrap>
+  )
+}
+
+RollBackIcon.propTypes = {
+  color: PropTypes.string,
+  width: PropTypes.string,
+}
+
+export const QuestionIcon: FunctionComponent<IconProps> = ({ color, width = '16' }): React.ReactElement => {
```
Comment (alfetopito) [inline src/components/icons/index.tsx:36]: Same point as the copy icon. There is already another question icon already used in the project

File: src/custom/pages/App/AppMod.tsx [hunk 4]
```diff
@@ -82,13 +100,22 @@ const Marginer = styled.div`
...
 
 export default function App(props?: { children?: ReactNode }) {
+  const [bgBlur, setBgBlur] = useState(false)
+  const location = useLocation()
+  useEffect(() => {
+    if (location.pathname.length > 1 && location.pathname !== '/swap') {
+      setBgBlur(true)
+    } else {
+      setBgBlur(false)
+    }
```
Comment (alfetopito) [inline src/custom/pages/App/AppMod.tsx:110]: Refactoring suggestion:

```suggestion
    setBgBlur(location.pathname.length > 1 && location.pathname !== '/swap')
```

Although, not sure should keep this logic. I'll wait for everyone else's input regarding the blur on swap page.

File: src/custom/pages/Profile/index.tsx [hunk 5]
```diff
@@ -1,87 +1,148 @@
...
+              <Txt fw={'bold'}>{label} ago</Txt>
+            </Txt>
+          </Wrap>
+        </Wrap>
+        <StyledWrap>
+          <Txt secondary fw={'bold'}>
+            Your referral url
+          </Txt>
+          <Wrap alignItems={'center'} justifyCont={'center'}>
+            <Txt onClick={() => copyToClipboard()} fs={14} txtAlign={'center'}>
```
Comment (alfetopito) [inline src/custom/pages/Profile/index.tsx:50]: There's a `<Copy>` component for that you can use. 
It's https://github.com/gnosis/cowswap/blob/7500644d17d8021fed99f218abf6582fde2ac7a1/src/custom/components/Copy/CopyMod.tsx#L31. 
Import it as `import Copy from 'components/Copy'`

File: src/custom/theme/baseTheme.tsx [hunk 6]
```diff
@@ -53,6 +53,9 @@ export function colors(darkMode: boolean): Colors {
     blueShade2: '#011e34',
     border: darkMode ? '#3a3b5a' : 'rgb(58 59 90 / 10%)',
     border2: darkMode ? '#254F83' : '#afcbda',
+    cardBorder: darkMode ? '#021E34' : 'rgba(255, 255, 255, 0.5)',
+    cardShadow21: darkMode ? '#4C7487' : '#FFFFFF',
+    cardShadow22: darkMode ? 'rgba(1, 10, 16, 0.15)' : 'rgba(11, 37, 53, 0.93)',
```
Comment (alfetopito) [inline src/custom/theme/baseTheme.tsx:58]: Any reason for using `21` then `22`?
Maybe you meant to have `1` and `2` ? Or no index and `2`?

File: src/custom/theme/cowSwapTheme.tsx [hunk 7]
```diff
@@ -111,9 +115,9 @@ function themeVariables(darkMode: boolean, colorsTheme: Colors) {
       `,
     },
     appBody: {
-      boxShadow: `4px 4px 0px ${colorsTheme.black}`,
-      borderRadius: '16px',
-      border: `3px solid ${colorsTheme.black}`,
+      boxShadow: `none`,
+      borderRadius: '1rem',
```
Comment (alfetopito) [inline src/custom/theme/cowSwapTheme.tsx:119]: I suspect @biocom will say: "use `px` instead of `rem` for consistency with the rest of the app"

**Review by alfetopito — COMMENTED** [review]
Looking good so far!

I left comments regarding the code, with a few overlaps with Elena's comments.
I'll ping the rest of the team to double check that part.

Now going over Elena's points:

1. Bottom elements missing: 
Agree, they should be visible. Probably need z-index adjustment or something?
2. Design is more blurred than implementation: 
We can make it more blurry, but I find it ok as is.
3. Swap page is not blurred while everything else is: 
Good point, we have to decide on the behaviour here.
I'm not that concerned for not being blurred on the swap page. I even makes sense, to have the main page not blurred, and on other pages you add a blur to indicate "secondary" or something.

My concern here is not having the swap border:
| before | now |
| - | - |
screenshot_2021-09-02_13-28-19 | screenshot_2021-09-02_13-17-24

I'm not saying we should add it back, just saying we probably need some way to emphasize the swap form.
Designers?

4. `Style` errors on the console: Agree
5. Copy icon wrapping: Agree
6. Copy icon style/behaviour: Agree
7. Cowgame score on mobile:
Just checked and the same happens on PROD, so not an issue introduced here.
8. Profile page width larger than design:
Not a big deal to me, I'll leave this up to the designers
9. Text colors on dark mode:
Same as previous

Regarding mobile points:

1. No announcement visible on top: 
Same as point `1` above, probably need to adjust z-index
2. No all items viewable in a single page: 
Agree, it could be make slight smaller and reduce the margin between cards


Regarding global question:

1. Different colors between design and current backgrounds. Also there's transparency in the design:
Hawk eyes! Another one that I leave up to the designers

File: src/custom/pages/App/AppMod.tsx [hunk 8]
```diff
@@ -35,14 +35,32 @@ import ReferralLinkUpdater from 'state/affiliate/updater'
...
   min-height: 100vh;
   overflow-x: hidden;
+  &:after {
+    content: '';
+    position: fixed;
+    top: 0;
+    left: 0;
+    bottom: 0;
+    right: 0;
+    filter: blur(5px);
```
Comment (alongoni) [inline src/custom/pages/App/AppMod.tsx:58]: Maybe:
filter: blur(10px);
cc @biocom?

**Review by alongoni — COMMENTED** [review]
Great job Maria!

File: src/assets/styles/styled.ts (hunk 1 above)
Comment (W3stside) [reply src/assets/styles/styled.ts:205]: yes please review the theme inside custom/theme and src/theme

File: src/assets/styles/styled.ts [hunk 9]
```diff
@@ -0,0 +1,268 @@
...
+      disabled?: boolean
+    }
+  >
+>`
+  display: inline-flex;
+  text-align: ${(props) =>
+    props.txtAlign ? (typeof props.txtAlign !== 'string' ? props.txtAlign[0] : props.txtAlign) : 'left'};
+  font-size: ${(props) => (props.fs ? (typeof props.fs !== 'number' ? props.fs[0] / 16 : props.fs / 16) : 0.87)}rem;
+  opacity: ${(props) => props.secondary && '0.5'};
+  line-height: ${(props) =>
```
[the remark this answers is not in the document]
Comment (W3stside) [reply src/assets/styles/styled.ts:61]: Hi @maria-vslvn yes this css file is much too overcomplicated imho.

Why add so many customisable props? why not use inheritance on a base component or simply create ones you need for the job? While i think it's nice to aim to make something generic and highly customisable, in this case the code is very difficult to read and overly verbose.

i would stick with smaller, more simple components here please.

**Review by anxolin — COMMENTED** [review]
Nice to see the profile page!

In general, agree with most comments. Just a couple of comments. I feel needs a bit of refinement in general. 

>> Swap page is not blurred while everything else is:
> Good point, we have to decide on the behaviour here.
> I'm not that concerned for not being blurred on the swap page. I even makes sense, to have the main page not blurred, and > on other pages you add a blur to indicate "secondary" or something.

Agree with @alfetopito . I find it ok, to have the main page non-blured and content pages with the effect.

Regarding the images, they look not very good quality. Not sure if it's the size, resolution, or small details.

Also, are the copy button, has a background only in light mode. And doesn't have any hover effect or cursor. 

image

Since it's the only button in the page, I wonder if it should look like the other buttons? 
image

The copy URL also should be dynamic

image

File: src/custom/theme/cowSwapTheme.tsx (hunk 7 above)
Comment (biocom) [reply src/custom/theme/cowSwapTheme.tsx:119]: Yes.

File: src/assets/styles/styled.ts (hunk 1 above)
Comment (biocom) [reply src/assets/styles/styled.ts:205]: Commented elsewhere as well. We use this format:

```
${({ theme }) => theme.mediaWidth.upToSmall`
  // stuff
`};
```

File: src/assets/styles/styled.ts (hunk 9 above)
[the remark this answers is not in the document]
Comment (biocom) [reply src/assets/styles/styled.ts:61]: Please simplify the code. Use px's instead of rem.

**Comment by biocom** [comment]
Here my review:
- For the 'click to copy' please see the sidebar click logic/style. Would be good to match the style + behavior (e.g. hover cursor).
- The Emoji images are intended to be actual emoji's. So use the actual emoji unicode characters -> 👨‍🌾🐮 etc.
- The blur effect was intended to be on all pages, except the Swap page. Primarily to function as a content page differentiator.
- For now don't modify the main Swap container's styles.

**Comment by elena-zh** [comment]
> Cowgame score on mobile:
> Just checked and the same happens on PROD, so not an issue introduced here.
 My bad.. I have created a separate issue for this case #1388

**Comment by elena-zh** [comment]
I think, I missed one important thing to mention: Profile page is does not correspond to the mock-up to a not connected user: 
 - Connect to a wallet button should be added to the bottom of the section. The button should be enabled and should call the connection modal
 - there should not be 'Last updated' field near the page header
image

**Comment by elena-zh** [comment]
As for 'Style' issue in the console, I can face it prod, but not so often as here. 
However, I created a separate issue for this case #1389

File: src/custom/pages/App/AppMod.tsx (hunk 8 above)
Comment (biocom) [reply src/custom/pages/App/AppMod.tsx:58]: 20px is better.

**Comment by alongoni** [comment]
hey Maria! Looks great. Some comments:
1)  The title of the page "Profile overview" is 32px on mobile. Maybe we should decreases it to 24px. It could affect all pages.
image
2) [Titles color on dark-mode](https://www.figma.com/file/bcoZf5RIgVKDItdOaTZq18/GP-Swap?node-id=2647%3A289). Titles should have #C5DAEF (or the equivalent color variable). Also I've made some little colors adjustments in light mode in order to have more consistent between light and dark mode (see comments on Figma).

**Comment by alfetopito** [comment]
Looking quite nice!

Regarding the swap page though, I think it shouldn't have a transparent background
screenshot_2021-09-20_15-17-00
In my view doesn't look as good as other pages since the background is not blurred.

Also, please don't forget to address the comments and mark as resolved the ones you already handled.

**Comment by biocom** [comment]
Agreed with @alfetopito 's comments. I'd leave the Swap page/container as is.

**Comment by elena-zh** [comment]
I have retested changes and here is the list of remaining issues to fix (besides mentioned above ones):
1. Page components are shifted to a not connected user. Moreover, 'Referral' section is not displayed in a mobile view now: 
not connected
not connected - missing

2. Last updated field should be removed from a profile page when a user is not connected
image

3. Connect to a wallet button should not present when a user is connected to a wallet
connect

4. 'Connect to a wallet' button style is different from the button's style at the top of the page.
image
I would reuse it, IMO
re-use
re-use2

5. Copy icon has a circle with bg color, however, original icon from the side bar does not have one: 
copy bg
copy bg2

6. Copy icon should be bottom-aligned in a mobile view
copy to bottom

7. Still, announcement area is empty at the top of the page
8. 'Style' error in console is not fixed
ыенду
9. Still, it would be nice to use smaller elements in a mobile view
10. Microsoft icons look.. strange =) However, I'm not sure whether we should fix it. Just to mention
microsoft icons

File: src/components/icons/index.tsx (hunk 2 above)
Comment (maria-vslvn) [reply src/components/icons/index.tsx]: agree, replaced with existing component

File: src/components/icons/index.tsx (hunk 3 above)
Comment (maria-vslvn) [reply src/components/icons/index.tsx:41]: agree, replaced with existing component

File: src/custom/pages/Profile/index.tsx (hunk 5 above)
Comment (maria-vslvn) [reply src/custom/pages/Profile/index.tsx]: applied

File: src/custom/theme/baseTheme.tsx (hunk 6 above)
Comment (maria-vslvn) [reply src/custom/theme/baseTheme.tsx:58]: yes, at first i thought about using indexes and forgot to remove it later. 2-index removed

File: src/assets/styles/styled.ts [hunk 10]
```diff
@@ -0,0 +1,268 @@
...
+  flexRow?: any
+  sm?: any
+  md?: any
+  lg?: any
+  xl?: any
+  stretch?: boolean
+}
+
+// with font size convert to rems
+export const Txt = styled.div<
```
[the remark this answers is not in the document]
Comment (maria-vslvn) [reply src/assets/styles/styled.ts:5]: agree, i used h3

File: src/custom/pages/Profile/index.tsx [hunk 11]
```diff
@@ -1,87 +1,148 @@
...
+                  <QuestionIcon />
+                </Wrap>
+                <Wrap width={'100%'}>
+                  <Wrap flexDir={'column'} alignItems={'center'} grow={'1'}>
+                    {/*temp pics solution  */}
+                    <img
+                      style={{ margin: ' .625rem auto' }}
+                      src="https://emojipedia-us.s3.dualstack.us-west-1.amazonaws.com/thumbs/72/apple/118/handshake_1f91d.png"
+                      alt={'handshake_1f91d'}
+                    />
```
[the remark this answers is not in the document]
Comment (maria-vslvn) [reply src/custom/pages/Profile/index.tsx]: done

File: src/custom/pages/Profile/index.tsx [hunk 12]
```diff
@@ -1,87 +1,148 @@
...
+                    <img
+                      style={{ margin: ' .625rem auto' }}
+                      src="https://emojipedia-us.s3.dualstack.us-west-1.amazonaws.com/thumbs/72/apple/118/money-with-wings_1f4b8.png"
+                      alt={'money-with-wings'}
+                    />
+                    <Txt fs={[26, 21]} fw={'bold'}>
+                      -
+                    </Txt>
+                    <Txt fs={14} secondary margin={'.5rem 0'}>
+                      Referrals Volume
```
[the remark this answers is not in the document]
Comment (maria-vslvn) [reply src/custom/pages/Profile/index.tsx]: done

**Comment by maria-vslvn** [comment]
@elena-zh 
5. https://www.figma.com/file/bcoZf5RIgVKDItdOaTZq18/GP-Swap?node-id=2647%3A289
but there is a bg in mockups, so i added
8. cant see style ers in console, could you please provide repro steps?
9. made as small as could, you can give me any more advices on what else to make smaller
10. that's just how they look ¯\_(ツ)_/¯

**Comment by alongoni** [comment]
> @elena-zh
> 5. https://www.figma.com/file/bcoZf5RIgVKDItdOaTZq18/GP-Swap?node-id=2647%3A289
> but there is a bg in mockups, so i added
> 8. cant see style ers in console, could you please provide repro steps?
> 9. made as small as could, you can give me any more advices on what else to make smaller
> 10. that's just how they look ¯_(ツ)_/¯

Great @maria-vslvn  thanks!
9) I think @elena-zh refers to this case, right? (medium-screens). Maybe we can change the breakpoint and maintain the desktop appearance for medium devices.
Screenshot from 2021-09-23 10-39-56
In small screens looks great!
10) The icons displayed may depend on the operating system. I see this icons on Chrome using Linux. Maybe we need to export them in svg to have consistency across all platforms.
Screenshot from 2021-09-23 10-55-22

File: src/assets/styles/styled.ts (hunk 9 above)
[the remark this answers is not in the document]
Comment (maria-vslvn) [reply src/assets/styles/styled.ts]: done

File: src/assets/styles/styled.ts (hunk 1 above)
Comment (maria-vslvn) [reply src/assets/styles/styled.ts]: done

**Comment by maria-vslvn** [comment]
@elena-zh  @alongoni  added the upd about decreasing the brekpoints

**Comment by elena-zh** [comment]
Thanks!!
Some new issues + some unfixed: 
1. Still missing the Black boarder on the Swap form
black border
2. Profile page is missing in the current PR. As well as 'About' section is removed from the menu
profile
3. Connection modal is broken: this is how it looks like now
connect dark
connect light
**Expected**: 
expected dark
expected lignt
4. When a screen width is less than 375 px, elements exceed screen area. Look: swap page is adjusted when it is displayed on the 320 px width, Profile page - not
profile - small screen
swap
5. Moreover, elements are shifted when copy referral link
https://user-images.githubusercontent.com/70885163/134530635-842a683d-7b5c-4771-bfcc-e7d20a5698a0.mp4
6. Copy icon still has background circle
 copy bg
copy bg2
7. It would be nice to move copy icon to be displayed right after URL . Or, if it impossible, to show it at the left pot corner.. @alongoni , @alfetopito , WDYT?
8. Tooltips: according to the discussion https://github.com/gnosis/cowswap/issues/958#issuecomment-906314157 we do not need tooltips, I guess. @alfetopito , correct?

Thanks

**Comment by alfetopito** [comment]
Replying to Elena's comments:

1. Yes, that's still missing. Swap UI should not be changed
2. Well, this is a constant change that has to be disabled on every release. I think it's fine not having it until we are ready to release the profile page, otherwise it might get removed again
3. Yep, should be like you said
4. Yep. For that either fix the base container to the cards minimum size or make the card container float. What's important is to not let the inner cards go over the outer container
5. Yes, it shifts a little, although I consider this a minor thing
6. Yep, should match the design
7. I don't get what you mean here Elena. The copy icon is already after the URL, no?
8. Tooltips can be added as a separated PR/issue

File: src/custom/pages/Profile/styled.tsx [hunk 13]
```diff
@@ -0,0 +1,136 @@
+import styled from 'styled-components'
+import Page, { GdocsListStyle, Title } from 'components/Page'
+import * as CSS from 'csstype'
+
+export const Wrapper = styled(Page)`
+  ${GdocsListStyle}
+
+  max-width: 910px;
+  min-height: auto;
+  padding-top: 1rem;
```
Comment (alfetopito) [inline src/custom/pages/Profile/styled.tsx:10]: Use px instead of rem :)

**Review by alfetopito — COMMENTED** [review]
Code is much better, great improvement!
Minor comment regarding the units

**Comment by elena-zh** [comment]
> > It would be nice to move copy icon to be displayed right after URL . Or, if it impossible, to show it at the left pot corner.. @alongoni , @alfetopito , WDYT?

> I don't get what you mean here Elena. The copy icon is already after the URL, no?

@alfetopito , I'm sorry, I forgot to attach the screenshot:
move copy

**Comment by alongoni** [comment]
hey @maria-vslvn 
Regarding 6) I've[ update the design](https://www.figma.com/file/bcoZf5RIgVKDItdOaTZq18/GP-Swap?node-id=1989%3A2) and remove the background for [copy button](https://feathericons.com/?query=copy).

**Comment by maria-vslvn** [comment]
1. fixed
2.  - 
3.  the modal incorrect look connected to the Connect button component reusability. if i should try fix it in this pr or use some separat button for that, wdyt @elena-zh @ramirotw ?
4. fixed
5. it was how it worked previously, if i should fix it here let me know or maybe we should create a ticket in order to avoid a lot of code in one pr and simplify its review
6. fixed

File: src/custom/pages/App/AppMod.tsx (hunk 4 above)
Comment (anxolin) [reply src/custom/pages/App/AppMod.tsx:113]: @maria-vslvn i think Leandro suggestion is good

**Review by anxolin — COMMENTED** [review]
> 3. the modal incorrect look connected to the Connect button component reusability. if i should try fix it in this pr or use some separat button for that, wdyt @elena-zh @ramirotw ?

That's fine with me, since this error doesn't affect to the main connect button and this page is hidden, is see no harm in merging and doing it in a different PR.

I know why this is happening, and I will create a small PR fixing the issue just for giving the direction.

**Comment by anxolin** [comment]
The PR i mentioned i my previous PR: @alfetopito @ramirotw could someone in protofire work on this? https://github.com/gnosis/cowswap/pull/1477

**Comment by elena-zh** [comment]
@maria-vslvn , great job!
I'm not able to test changes in this PR, as I get 404 error on opening it, but i was able to test changes in #1477 pr. 
1. The black boarder is back in the Swap page, however, it is back to the About/FAQ/Terms pages, As far as I understand, only Swap page should have black boarder on its area. All the rest pages (Profile/About/FAQ/Terms) should look the same 
2. I have created a separate issue for the case 5: #1478
3. Still I'd vote for moving 'Copy' icon closer to the referral link 
image
4. Btw, @alfetopito , @alongoni , should we have some kind if animation when press on the 'Last updated' icon? I know that it is not related to this PR, but just a question.
image

**Comment by elena-zh** [comment]
I have retested changes in the current PR after deployment error fix. All the mentioned above points are valid for this PR.

**Comment by alongoni** [comment]
> 4. Btw, @alfetopito , @alongoni , should we have some kind if animation when press on the 'Last updated' icon? I know that it is not related to this PR, but just a question.
>    image

I think it is just an icon with automatic update. Maybe we can add a transition each time it runs.

**Comment by alfetopito** [comment]
>1. The black boarder is back in the Swap page, however, it is back to the About/FAQ/Terms/404 pages, As far as I understand, only Swap page should have black boarder on its area. All the rest pages (Profile/About/FAQ/Terms/404) should look the same

Yep. But I believe we are in a state that can be merged and address the remaining points in follow up PRs
 
>2. I have created a separate issue for the case 5: [Mobile: Profile elements are shifted when copy referral link #1478](https://github.com/gnosis/cowswap/issues/1478)
> 
>3. Still I'd vote for moving 'Copy' icon closer to the referral link
>        image

Agree. Maybe that'll help solve the issue with mobile and the text shifting mentioned on `3`. Then again, another PR/issue

>4. Btw, @alfetopito , @alongoni , should we have some kind if animation when press on the 'Last updated' icon? I know that it is not related to this PR, but just a question.
>image

No, this is intended to be a static icon. Maybe in a future version we could have that, but right now the button is not intended to be clickable.

**Review by alfetopito — APPROVED** [review]
I quite like the result and I believe this is already "merge-able"

There are a few things to address yet but they can be addressed in smaller PRs.
Also, some of the issue are not entirely UI specific, which is out of the scope of Maria's task.

With that in mind will be easier for reviewers to comment

When CSS/design people (@alongoni @biocom) are happy let's merge it.

@maria-vslvn in the mean time, you can start addressing the pending issues in PRs pointing to this branch.

**Comment by biocom** [comment]
Minor thing i missed:
- The profile container misses this setting `backdrop-filter: blur(50px);`

Otherwise OK to go as is.

**Comment by elena-zh** [comment]
I have created separate issues for moving copy closed to the link (#1480) and boarder color fix (#1481)

**Review by elena-zh — APPROVED** [review]

**Review by alongoni — APPROVED** [review]

**Pull request merged**